### PR TITLE
Adding takingoff, landing, on ground state

### PR DESCRIPTION
**Pull request by shrit**
This patch allows users to distinguish the detailed
state of the UAVs in real-time. IN AIR state can be
ambiguous since the UAVs in general or quadrotors in
particular can take off and land.

Signed-off-by: Omar Shrit <shrit@lri.fr>

### CODE REVIEW

**Comment by shrit** [comment]
Hello,
Thanks for your fast reply.
I agree with you about using one function and enum instead of 4 functions.
I will rename it to `landed_state` instead of `in_air`.
I will create a enum for the 4 states. 
To give more details, I do simulation with several quadcopters at the same time. When I send landing I need a blocking function to stop code execution until all quadcopters have landed successfully. In this case `in_air()` function do the job well. I can wait as long as the quadcopter are not on the ground.
The problem arise from taking off, for example, if I need a quadcopter to go up 20 meters, I also need a blocking function that stop code execution until the takeoff process has finished.

**Comment by JonasVautherin** [comment]
Sounds good to me. Note that for consistency, [`in_air_async`](https://github.com/mavlink/MAVSDK/blob/develop/src/plugins/telemetry/include/plugins/telemetry/telemetry.h#L688) should probably be updated as well.

Let's wait for @julianoes' opinion on that, I would say!

**Comment by shrit** [comment]
Agreed, let's wait for @julianoes. I have finished the landed state.

File: src/plugins/telemetry/include/plugins/telemetry/telemetry.h [hunk 1]
```diff
@@ -226,6 +226,16 @@ class Telemetry : public PluginBase {
      */
     static std::string flight_mode_str(FlightMode flight_mode);
 
+    /**
+     *@brief LandedState.
+     * Enumeration of landed detector states
+     * For more information, check out MavLink messages in
+     * https://mavlink.io/en/messages/common.html
+     */
+    enum class LandedState { UNDEFINED, ON_GROUND, IN_AIR, TAKEOFF, LANDING };
```
[the remark this answers is not in the document]
Comment (julianoes) [reply src/plugins/telemetry/include/plugins/telemetry/telemetry.h]: What about `TAKING_OFF`?

File: src/plugins/telemetry/telemetry.cpp [hunk 2]
```diff
@@ -145,6 +145,33 @@ bool Telemetry::in_air() const
...
+
+std::string Telemetry::landed_state_str(LandedState landed_state)
+{
+    switch (landed_state) {
+        case LandedState::ON_GROUND:
+            return "On_ground";
+        case LandedState::IN_AIR:
+            return "In air";
+        case LandedState::TAKEOFF:
+            return "Takingoff";
```
Comment (julianoes) [inline src/plugins/telemetry/telemetry.cpp]: And here to `Taking off`.

File: src/plugins/telemetry/include/plugins/telemetry/telemetry.h [hunk 3]
```diff
@@ -226,6 +226,16 @@ class Telemetry : public PluginBase {
      */
     static std::string flight_mode_str(FlightMode flight_mode);
 
+    /**
+     *@brief LandedState.
+     * Enumeration of landed detector states
+     * For more information, check out MavLink messages in
+     * https://mavlink.io/en/messages/common.html
```
Comment (julianoes) [inline src/plugins/telemetry/include/plugins/telemetry/telemetry.h]: This reference to MAVLink should not be necessary.

**Comment by shrit** [comment]
Done, @julianoes, @JonasVautherin, I think you can merge now.

**Comment by julianoes** [comment]
Travis complains about one missing docstring:
```
Generating code for file plugin_bas/home/user/DronecodeSDK/tools/docs/install/include/mavsdk/plugins/telemetry/telemetry.h:235: warning: Member landed_state_str(LandedState landed_state) (function) of class mavsdk::Telemetry is not documented.
```

And Jenkins shows a weird abort, not sure if it's related
```
integration_tests_runner: tpp.c:84: __pthread_tpp_change_priority: Assertion `new_prio == -1 || (new_prio >= fifo_min_prio && new_prio <= fifo_max_prio)' failed.

Aborted (core dumped)
```

**Comment by shrit** [comment]
I have pulled before committing to avoid merge conflicts

**Comment by shrit** [comment]
@JonasVautherin, I have commited, pulled, then I have merged, Then I have added more commits and so on.

**Comment by shrit** [comment]
if it is necessary, I think I can do revert for the fcbee00 commit. What do you think @JonasVautherin ?

File: src/plugins/telemetry/telemetry.cpp [hunk 4]
```diff
@@ -145,6 +145,33 @@ bool Telemetry::in_air() const
...
+void Telemetry::landed_state_async(landed_state_callback_t callback)
+{
+    _impl->landed_state_async(callback);
+}
+
+std::string Telemetry::landed_state_str(LandedState landed_state)
+{
+    switch (landed_state) {
+        case LandedState::ON_GROUND:
+            return "On_ground";
```
Comment (julianoes) [inline src/plugins/telemetry/telemetry.cpp]:
```suggestion
            return "On ground";
```

File: src/plugins/telemetry/telemetry.cpp [hunk 5]
```diff
@@ -145,6 +145,33 @@ bool Telemetry::in_air() const
...
+        case LandedState::IN_AIR:
+            return "In air";
+        case LandedState::TAKING_OFF:
+            return "Taking_off";
```
Comment (julianoes) [inline src/plugins/telemetry/telemetry.cpp]:
```suggestion
            return "Taking off";
```

**Comment by JonasVautherin** [comment]
> if it is necessary, I think I can do revert for the fcbee00 commit. What do you think @JonasVautherin ?

This is not how it works. I just merged `develop` into your branch, and it is fine now.

The way it works is that you should keep your fork up-to-date ([see docs here](https://help.github.com/en/articles/syncing-a-fork)). You should never have to do anything other than `merge` or `rebase` the main branch into your working branch.

Thanks for the contribution! :rocket:

**Pull request merged**